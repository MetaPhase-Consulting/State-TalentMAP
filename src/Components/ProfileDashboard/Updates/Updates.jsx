import React, { Component } from 'react';
import FA from 'react-fontawesome';
import SectionTitle from '../SectionTitle';
import CheckboxList from '../../BidderPortfolio/CheckboxList';
import EditButtons from '../../BidderPortfolio/EditButtons';

class Updates extends Component {
  constructor(props) {
    super(props);
    this.enableEdit = this.enableEdit.bind(this);
    this.disableEdit = this.disableEdit.bind(this);
    this.state = {
      isEditable: false,
    };
  }
  enableEdit() {
    this.setState({ isEditable: true });
  }
  disableEdit() {
    this.setState({ isEditable: false });
  }
  render() {
    const { isEditable } = this.state;
    return (
      <div className="usa-grid-full profile-section-container updates-container">
        <div className="usa-grid-full section-padded-inner-container">
          <div className="usa-width-one-whole">
            <SectionTitle title="Bidder Classifications" icon="tasks" />
          </div>
          <div className="usa-width-one-whole">
            <CheckboxList isDisabled={!isEditable} id="updates" />
          </div>
        </div>
        <div className="section-padded-inner-container small-link-container view-more-link-centered">
          { !isEditable && <button className="unstyled-button" onClick={this.enableEdit}><FA name="pencil" /> Edit Classifications</button> }
          { isEditable && <EditButtons initialShowSave onChange={this.disableEdit} /> }
        </div>
      </div>
    );
  }
}

export default Updates;
