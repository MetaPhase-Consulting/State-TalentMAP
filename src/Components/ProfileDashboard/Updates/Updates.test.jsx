import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Updates from './Updates';

describe('UpdatesComponent', () => {
  it('is defined', () => {
    const wrapper = shallow(<Updates />);
    expect(wrapper).toBeDefined();
  });

  it('renders no button on mount when showEditButton is false', () => {
    const wrapper = shallow(<Updates />);
    expect(wrapper.find('button').length).toBe(0);
  });

  // it('renders one button on mount when showEditButton is true ', () => {
  //   const wrapper = shallow(<Updates />);
  //   expect(wrapper.find('button').length).toBe(1);
  //   expect(wrapper.find('EditButtons').exists()).toBe(false);
  // });

  // it('renders two buttons when the Edit button is clicked', () => {
  //   const wrapper = shallow(<Updates />);
  //   wrapper.find('button').simulate('click');
  //   expect(wrapper.find('button').exists()).toBe(false);
  //   expect(wrapper.find('EditButtons').exists()).toBe(true);
  // });

  // it('renders one button when enableEdit() is called and then disableEdit() is called', () => {
  //   const wrapper = shallow(<Updates />);
  //   const instance = wrapper.instance();
  //   instance.enableEdit();
  //   instance.disableEdit();
  //   expect(wrapper.find('button').length).toBe(1);
  //   expect(wrapper.find('EditButtons').exists()).toBe(false);
  // });


  it('matches snapshot', () => {
    const wrapper = shallow(<Updates />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
