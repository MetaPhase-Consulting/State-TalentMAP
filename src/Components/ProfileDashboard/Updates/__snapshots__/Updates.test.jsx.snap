// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdatesComponent matches snapshot 1`] = `
<div
  className="usa-grid-full profile-section-container updates-container"
>
  <div
    className="usa-grid-full section-padded-inner-container"
  >
    <div
      className="usa-width-one-whole"
    >
      <PositionInformation
        className=""
        icon="tasks"
        small={false}
        title="Updates"
      />
    </div>
    <div
      className="usa-width-one-whole"
    >
      <CheckboxList
        id="updates"
        isDisabled={true}
      />
    </div>
  </div>
  <div
    className="section-padded-inner-container small-link-container view-more-link-centered"
  >
    <button
      className="unstyled-button"
      onClick={[Function]}
    >
      <FontAwesome
        name="pencil"
      />
       Edit Updates
    </button>
  </div>
</div>
`;
